src/Documentation/grains/transactions.md

```diff
@@ -3,82 +3,110 @@ layout: page
 
 Example:
 
 ``` csharp
 var builder = new SiloHostBuilder()
-    [...]
-    .AddAzureTableGrainStorageAsDefault(options => options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”)
-    .UseInClusterTransactionManager()
-    .UseAzureTransactionLog(options => options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”) 
-    .UseTransactionalState();
-
-var host = builder.Build();
-await host.StartAsync();
+    .AddAzureTableTransactionalStateStorage(“TransactionStore”, options =>
+    {
+        options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”);
+    })
+    .UseTransactions();
 ```
 
+For development purposes, if transaction specific storage is not available for the data store you need, an `IGrainStorage` implementation may be used instead.
+For any transactional state which does not have a store configured for it, transactions will attempt to fail over to grains storage using a bridge.
+Accessing transactional state via a bridge to grains storage will be less efficient and is not a pattern we intend to support long term, hence the recommendation that this only be used for development purposes.
+
 ## Programming Model
 
-For a grain to support transactions, transactional operations on a grain must be marked as being part of a transaction using the “Transaction” attribute.  Calls can be marked as “Required”, meaning the call can take part in a transaction among multiple other grains or as “RequiresNew” indicating that it will already start its own transaction.
+### Grain Interfaces
 
-Example:
+For a grain to support transactions, transactional operations on a grain interface must be marked as being part of a transaction using the “Transaction” attribute.
+The attribute needs indicate how the grain call behaves in a transactional environment.
+It indicates this via the transaction options below:
+
+- `TransactionOption.Create` - Call is transactional and will always create a new transaction context, even if called within an existing transaction context.
+- `TransactionOption.Join Call` - Logic is transactional but can only be called within the context of an existing transaction.
+- `TransactionOption.CreateOrJoin` - Call is transactional. If called within the context of a transaction, it will use that context, else it will create a new context.
+- `TransactionOption.Suppress` - Call is not transactional but can be called from within a transaction. If called within the context of a transaction, the context will not be passed to the call.
+- `TransactionOption.Supported` - Call is not transactional but supports transactions. If called within the context of a transaction, the context will be passed to the call.
+- `TransactionOption.NotAllowed` - Call is not transactional and cannot be called from within a transaction. If called within the context of a transaction, it will throw a not supported exception.
```

Comment:
Perhaps specify the precise exception here, `System.NotSupportedException` or just `NotSupportedException`

Reply:
I went with the less verbose option, to keep it like the other verbiage.

- `TransactionOption.NotAllowed` - Call is not transactional and cannot be called from within a transaction. If called within the context of a transaction, it will throw a `NotSupportedException`.